packages/manager/src/features/CloudPulse/Widget/CloudPulseWidget.test.tsx

```diff
@@ -0,0 +1,189 @@
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import {
+  cloudPulseMetricsResponseDataFactory,
+  widgetFactory,
+} from 'src/factories';
+import * as CloudPulseWidgetUtils from 'src/features/CloudPulse/Utils/CloudPulseWidgetUtils';
+import { formatPercentage } from 'src/utilities/statMetrics';
+import { renderWithTheme } from 'src/utilities/testHelpers';
+
+import { CloudPulseWidget } from './CloudPulseWidget';
+
+import type { CloudPulseWidgetProperties } from './CloudPulseWidget';
+
+const props: CloudPulseWidgetProperties = {
+  additionalFilters: [],
+  ariaLabel: 'CPU Utilization',
+  availableMetrics: {
+    available_aggregate_functions: ['min', 'max', 'avg'],
+    dimensions: [],
+    label: 'CPU utilization',
+    metric: 'system_cpu_utilization_percent',
+    metric_type: 'gauge',
+    scrape_interval: '2m',
+    unit: 'percent',
+  },
+  duration: { unit: 'min', value: 30 },
+  isJweTokenFetching: false,
+  resourceIds: ['1', '2'],
+  resources: [
+    {
+      id: '1',
+      label: 'test-1',
+    },
+    {
+      id: '2',
+      label: 'test-2',
+    },
+  ],
+  savePref: true,
+  serviceType: 'linode',
+  unit: '%',
+  widget: widgetFactory.build({
+    label: 'CPU Utilization',
+  }),
+};
+
+const queryMocks = vi.hoisted(() => ({
+  useCloudPulseMetricsQuery: vi.fn().mockReturnValue({}),
+}));
+
+vi.mock('src/queries/cloudpulse/metrics', async () => {
+  const actual = await vi.importActual('src/queries/cloudpulse/metrics');
+  return {
+    ...actual,
+    useCloudPulseMetricsQuery: queryMocks.useCloudPulseMetricsQuery,
+  };
+});
+
+queryMocks.useCloudPulseMetricsQuery.mockReturnValue({
+  data: cloudPulseMetricsResponseDataFactory.build(),
+  isError: false,
+  isLoading: false,
+  status: 'success',
+});
+
+const mockMetrics = {
+  average: 5.5,
+  last: 7.75,
+  length: 3,
+  max: 10,
+  total: 40,
+};
+const graphData = {
+  areas: [
+    {
+      color: '#1CB35C',
+      dataKey: 'test-1',
+    },
+  ],
+  dimensions: [],
+  legendRowsData: [
+    {
+      data: mockMetrics,
+      format: formatPercentage,
+      handleLegendClick: vi.fn(),
+      legendColor: 'blue',
+      legendTitle: 'Test',
+    },
+  ],
+  unit: '%',
+};
+vi.spyOn(CloudPulseWidgetUtils, 'generateGraphData').mockReturnValue(graphData);
+
+class ResizeObserver {
+  disconnect() {}
+  observe() {}
+  unobserve() {}
+}
+
+const mockUpdatePreferences = vi.fn();
+vi.mock('../Utils/UserPreference', () => ({
+  useAclpPreference: () => ({
+    updateWidgetPreference: mockUpdatePreferences,
+  }),
+}));
+
+describe('Cloud pulse widgets', () => {
+  window.ResizeObserver = ResizeObserver;
+  it('should render the widget with correct title and unit', () => {
+    const { getByText } = renderWithTheme(<CloudPulseWidget {...props} />);
```

Comment:
Since this is a single widget, you can validate the widget title, the filters like aggregation and interval filter everything here itself